File: test/python/src/quickstart.py
# HTTPState, https://httpstate.com/
# Copyright (C) Alex Morales, 2026

# Unless otherwise stated in particular files or directories, this software is free software.
# You can redistribute it and/or modify it under the terms of the GNU Affero
# General Public License as published by the Free Software Foundation, either
# version 3 of the License, or (at your option) any later version.

# pip install httpstate

import datetime
import httpstate
import threading

uuid = '58bff2fcbeb846958f36e7ae5b8a75b0'

print(datetime.datetime.now(datetime.UTC).isoformat(), 'httpstate', uuid)

httpstate.HTTPState('58bff2fcbeb846958f36e7ae5b8a75b0').on('change', lambda data : print(datetime.datetime.now(datetime.UTC).isoformat(), 'data', data))

# Not needed per se, only meant to keep the script alive
threading.Event().wait()

File: test/go/quickstart.go
// HTTPState, https://httpstate.com/
// Copyright (C) Alex Morales, 2026
//
// Unless otherwise stated in particular files or directories, this software is free software.
// You can redistribute it and/or modify it under the terms of the GNU Affero
// General Public License as published by the Free Software Foundation, either
// version 3 of the License, or (at your option) any later version.

// go mod init quickstart
// go get github.com/httpstate/httpstate/go@latest

package main

import (
	"fmt"
	"time"

	"github.com/httpstate/httpstate/go"
)

func main() {
	httpstate.New("58bff2fcbeb846958f36e7ae5b8a75b0", nil).On("change", func(result *httpstate.GetResult) {
		if result != nil {
			fmt.Println(time.Now().UTC().Format(time.RFC3339), "data", result.Data)
		}
	})

	// Not needed per se, only meant to keep the script alive
	select {}
}
